safehttp/plugins/csp/csp.go

```diff
@@ -166,6 +167,48 @@ func (f FramingPolicy) Serialize(nonce string) string {
 	return b.String()
 }
 
+// TrustedTypesPolicy policy can be used to create a new CSP policy which makes
+// dangerous web API functions secure by default.
+//
+// See https://web.dev/trusted-types for more info.
+type TrustedTypesPolicy struct {
+	// ReportURI controls the report-uri directive. If ReportUri is empty, no report-uri
+	// directive will be set.
+	ReportURI string
+	// PolicyNames controls allowed names of policies that can be created.
+	PolicyNames []string
+	// AllowDuplicates allows creating policies with a name that was already used.
+	AllowDuplicates bool
+}
+
+// Serialize serializes this policy for use in a Content-Security-Policy header
+// or in a Content-Security-Policy-Report-Only header. A nonce will be provided
+// to Serialize which can be used in 'nonce-{random-nonce}' values in directives.
+func (t TrustedTypesPolicy) Serialize(nonce string) string {
+	var b strings.Builder
+	b.WriteString("require-trusted-types-for 'script'")
+
+	if len(t.PolicyNames) > 0 {
+		b.WriteString("; trusted-types")
+		for _, name := range t.PolicyNames {
+			b.WriteString(" ")
+			b.WriteString(name)
+		}
+
+		if t.AllowDuplicates {
+			b.WriteString(" ")
+			b.WriteString("'allow-duplicates'")
+		}
+	}
+
+	if t.ReportURI != "" {
+		b.WriteString("; report-uri ")
+		b.WriteString(t.ReportURI)
+	}
```

Comment:
This block tends to be a bit too indented. Since we're not supporting named policies this will go away but in general please consider preserving the line of sight for the happy path and favor early returns: http://go/gotip/episodes/1

Reply:
Sounds good